packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,157 @@
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                Ok(value.to_vec())
+            }
+            fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+                Ok(value.to_vec())
+            }
+        })*
+    }
+}
+
+bytes_de!(for Vec<u8>, &Vec<u8>, [u8], &[u8]);
+
+macro_rules! string_de {
+    (for $($t:ty),+) => {
+        $(impl Deserializable for $t {
+            type Output = String;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                // FIXME?: Use `from_utf8_unchecked` for String, &str
+                String::from_utf8(value.to_vec())
+                    // FIXME: Add and use StdError utf-8 error From helper
+                    .map_err(|err| StdError::generic_err(err.to_string()))
+            }
+
+            fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+                // FIXME?: Use `from_utf8_unchecked` for String, &str
+                String::from_utf8(value.to_vec())
+                    // FIXME: Add and use StdError utf-8 error From helper
+                    .map_err(|err| StdError::generic_err(err.to_string()))
+            }
+        })*
+    }
+}
+
+string_de!(for String, &String, str, &str, Addr, &Addr);
+
+macro_rules! integer_de {
+    (for $($t:ty),+) => {
+        $(impl Deserializable for IntKey<$t> {
+            type Output = $t;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                Ok(<$t>::from_be_bytes(value.try_into()
+                    // FIXME: Add and use StdError try-from error From helper
+                    .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?))
+            }
+
+            fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+                Ok(<$t>::from_be_bytes(value.to_vec().try_into().unwrap()))
+            }
+        })*
+    }
+}
+
+integer_de!(for i8, u8, i16, u16, i32, u32, i64, u64, i128, u128);
+
+impl<T: Deserializable, U: Deserializable> Deserializable for (T, U) {
+    type Output = (T::Output, U::Output);
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        let t_len = u16::from_be_bytes(
+            value[..2]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let t = T::from_slice(&value[2..2 + t_len])?;
+        let u = U::from_slice(&value[2 + t_len..])?;
+
+        Ok((t, u))
+    }
+
+    fn from_vec(value: &Vec<u8>) -> StdResult<Self::Output> {
+        let t_len = u16::from_be_bytes(
+            value[..2]
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ) as usize;
+        let t = T::from_slice(&value[2..2 + t_len])?;
```

Comment:
Is range here required?
`&value[2..2 `

Reply:
This is just length prefixed deserialization. See `to_length_prefixed` and friends to get a better grasp at it. A way to improve on this would be to add a `from_length_prefixed` to encapsulate this, passing the number of slices.

Can be done in a follow-up.